### DISCUSSION TITLE
Lift Code

### DISCUSSION THREAD

**Discussion by ghost** [JSBSim-Team/jsbsim#645] [Q&A]
Where is all the code and such for applying lift to the aircraft? Is it all in the FGAerodynamics file? The file seems to talk most about it, but, it's rather hard to determine where a lot of the other files play in.

**Comment by seanmcleod** [comment]
So I assume you know that the aircraft modeler gets to specify a set of functions in the aircraft's FDM xml file for 3 force axes and 3 moment axes.

`FGAerodynamics` then runs through the 3 force axes and queries each function defined for the force the function returns and simply sums them all for each axis. It does the same thing for the 3 moment axes.

Summing the forces for each force axis:

https://github.com/JSBSim-Team/jsbsim/blob/76a4d4fb37e351714664119c61f2853ad23b7a15/src/models/FGAerodynamics.cpp#L185-L196

The same for each moment axis:

https://github.com/JSBSim-Team/jsbsim/blob/76a4d4fb37e351714664119c61f2853ad23b7a15/src/models/FGAerodynamics.cpp#L253-L263

The resulting forces are then applied at the aircraft's cg etc.

So I'm not sure what lift specific code you're thinking there should be.

**Comment by ghost** [comment]
I think this is what I'm looking for, more/less. I was looking for what applies the forces and makes it turn/change velocity as it would in real life.

**Comment by seanmcleod** [comment]
In addition to the aerodynamic forces and moments JSBSim will also calculate the propulsion forces and moments, ground forces from the gear etc. and then that combination is used to calculate the accelerations which are then integrated to get translational velocities and angular velocities which are then integrated again to get position and attitude states.

**Reply by ghost** [reply]
Is this all or mostly in the aerodynamics file?

**Comment by seanmcleod** [comment]
No. The aerodynamics source file just deals with calculating aerodynamic forces and moments.
